internal/pkg/cli/pipeline_delete.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/secretsmanager"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	termprogress "github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/progress"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	pipelineDeleteConfirmPrompt       = "Are you sure you want to delete pipeline %s from project %s?"
+	pipelineDeleteConfirmHelp         = "This will delete the deployment pipeline for the app(s) in the current project workspace."
+	pipelineSecretDeleteConfirmPrompt = "Are you sure you want to delete the source secret %s associated with pipeline %s?"
+	pipelineDeleteSecretConfirmHelp   = "This will delete the secret that stores the token associated with the source of your pipeline in the current project workspace."
```

Comment:
Based on some previous discussions with @bvtujo and @kohidave I think we wanted to keep the secret by default bc assuming they're keeping their same GitHub repo around, it can be annoying to delete a secret due to the "draining" period that secrets manager enforces. we could add the option to force-delete the secret so there isn't that waiting period, for ease of re-creation?